devlib/utils/android.py

```diff
@@ -377,12 +381,16 @@ def _background(self, command, stdout, stderr, as_root):
         return bg_cmd
 
     def _close(self):
-        AdbConnection.active_connections[self.device] -= 1
-        if AdbConnection.active_connections[self.device] <= 0:
+        lock, nr_active = AdbConnection.active_connections
+        with lock:
+            nr_active[self.device] -= 1
+            disconnect = nr_active[self.device] <= 0
+            del nr_active[self.device]
```

Comment:
Why is this moved here? Will this not delete the entry even if there are still connections left?

Reply:
Should be moved a line lower to be inside the `if` indeed